tests/Aspire.Hosting.Tests/ResourceExtensionsTests.cs

```diff
@@ -25,4 +25,102 @@ public void TryGetContainerImageNameReturnsCorrectFormatWhenShaNotSupplied()
         Assert.True(container.Resource.TryGetContainerImageName(out var imageName));
         Assert.Equal("grafana/grafana:10.3.1", imageName);
     }
+
+    [Fact]
+    public async Task GetEnvironmentVariableValuesAsyncReturnCorrectVariablesInRunMode()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+        var container = builder.AddContainer("elasticsearch", "library/elasticsearch", "8.14.0")
+         .WithEnvironment("discovery.type", "single-node")
+         .WithEnvironment("xpack.security.enabled", "true")
+         .WithEnvironment(context =>
+         {
+             context.EnvironmentVariables["ELASTIC_PASSWORD"] = "p@ssw0rd1";
```

Comment:
I think we are still going to get cred scan failures here, but we can try it.

Reply:
Will see. @radical was saying that they were in a suppression list.

Reply:
There are 2 cred scan checks, and my understanding is that only one of them respects the suppression list.